test/test-eth-trezor-keyring.js

```diff
@@ -694,4 +694,70 @@ describe('TrezorKeyring', function () {
       }
     });
   });
+
+  describe('getVendor', function () {
+    it('should call TrezorConnect.getFeatures if we dont have vendor name', async function () {
+      sinon
+        .stub(TrezorConnect, 'getFeatures')
+        .callsFake(() => Promise.resolve({}));
+
+      await keyring.getVendor();
+      assert(TrezorConnect.getFeatures.calledOnce);
+    });
+
+    it('should return null when getFeatures not success', async function () {
+      sinon
+        .stub(TrezorConnect, 'getFeatures')
+        .callsFake(() => Promise.resolve({}));
+
+      const vendor = await keyring.getVendor();
+      assert.equal(vendor, null);
+    });
+
+    it('should return trezor when getFeatures does not have minor_version or patch_version', async function () {
+      sinon.stub(TrezorConnect, 'getFeatures').callsFake(() =>
+        Promise.resolve({
+          success: true,
+          payload: {
+            vendor: 'trezor.io',
+          },
+        }),
+      );
+
+      const vendor = await keyring.getVendor();
+      assert.equal(vendor, 'trezor');
+    });
+
+    it('should return onekey when minor_version and patch_version is a special version', async function () {
+      sinon.stub(TrezorConnect, 'getFeatures').callsFake(() =>
+        Promise.resolve({
+          success: true,
+          payload: {
+            minor_version: 99,
+            patch_version: 99,
+            vendor: 'trezor.io',
+          },
+        }),
+      );
+
+      const vendor = await keyring.getVendor();
+      assert.equal(vendor, 'onekey');
+    });
+
+    it('should return onekey when vendor field is onekey.so', async function () {
+      sinon.stub(TrezorConnect, 'getFeatures').callsFake(() =>
+        Promise.resolve({
+          success: true,
+          payload: {
+            minor_version: 1,
+            patch_version: 1,
+            vendor: 'onekey.so',
+          },
+        }),
+      );
+
+      const vendor = await keyring.getVendor();
+      assert.equal(vendor, 'onekey');
+    });
```

Comment:
What do you think about adding a test that says that `TrezorConnect.getFeatures` is only called once even when `getVendor` is called more than once?

Reply:
Yes, I think your suggestion is necessary and have added this unit test. 👍